modules/apps/translation/translation-web/test/js/translate/Translate.js

```diff
@@ -0,0 +1,310 @@
+					sourceContentDir: 'ltr',
+					targetContent: '<p>mock summary</p>',
+					targetContentDir: 'ltr',
+					targetLanguageId: 'es_ES',
+				},
+			],
+			legend: 'Basic Information',
+		},
+		{
+			fields: [
+				{
+					editorConfiguration: {},
+					html: true,
+					id: 'infoField--content--',
+					label: 'Content',
+					multiline: true,
+					sourceContent: '<p>mock conent</p>',
+					sourceContentDir: 'ltr',
+					targetContent: '<p>mock conent</p',
+					targetContentDir: 'ltr',
+					targetLanguageId: 'es_ES',
+				},
+			],
+			legend: 'Content (Basic Web Content)',
+		},
+	],
+	portletId: 'mock_TranslationPortlet',
+	portletNamespace: '_mock_TranslationPortlet_',
+	publishButtonDisabled: false,
+	publishButtonLabel: 'Publish',
+	redirectURL: 'http://redirect-url',
+	saveButtonDisabled: false,
+	saveButtonLabel: 'Save as Draft',
+	sourceLanguageId: 'en_US',
+	sourceLanguageIdTitle: 'en-US',
+	targetLanguageId: 'es_ES',
+	targetLanguageIdTitle: 'es-ES',
+	translateLanguagesSelectorData: {
+		currentUrl: 'http://current-url',
+		sourceAvailableLanguages: ['en_US', 'es_ES'],
+		sourceLanguageId: 'en_US',
+		targetAvailableLanguages: [
+			'ar_SA',
+			'ca_ES',
+			'zh_CN',
+			'nl_NL',
+			'fi_FI',
+			'fr_FR',
+			'de_DE',
+			'hu_HU',
+			'ja_JP',
+			'pt_BR',
+			'es_ES',
+			'sv_SE',
+		],
+		targetLanguageId: 'es_ES',
+	},
+	translationPermission: true,
+	updateTranslationPortletURL: 'http://update-url',
+	workflowActions: {
+		PUBLISH: '1',
+		SAVE_DRAFT: '2',
+	},
+};
+
+const renderComponent = (props) => render(<Translate {...props} />);
+
+describe('Translate', () => {
+	Liferay.Util.sub.mockImplementation((langKey, ...args) =>
+		[langKey, ...args].join('-')
+	);
+
+	afterEach(cleanup);
+
+	it('renders with auto-translate enabled', () => {
+		const {asFragment} = renderComponent(baseProps);
+
+		expect(asFragment()).toMatchSnapshot();
+	});
+
+	it('renders auto-translate field button disabled when the sourceContent is empty', () => {
+		const {getByText} = renderComponent({
+			...baseProps,
+			infoFieldSetEntries: [
+				{
+					...baseProps.infoFieldSetEntries[1],
+					fields: [
+						{
+							...baseProps.infoFieldSetEntries[1].fields[0],
+							sourceContent: '',
+						},
+					],
+				},
+			],
+		});
+
+		expect(
+			getByText(
+				'auto-translate-x-field-' +
+					baseProps.infoFieldSetEntries[1].fields[0].label
+			).closest('button')
+		).toBeDisabled();
+	});
+
+	describe('when there is a valid server response', () => {
+		beforeEach(() => {
+			fetch.mockResponseOnce(
+				JSON.stringify({
+					fields: {
+						'infoField--content--': '<p>simulacro de contenido</p>',
+						'infoField--description--': '<p>resumen simulado</p>',
+						'infoField--title--': 'título simulado',
+					},
+					sourceLanguageId: 'en_US',
+					targetLanguageId: 'es_ES',
+				})
+			);
+		});
+
+		afterEach(() => {
+			fetch.resetMocks();
+		});
+
+		describe('click in the field auto-translate button', () => {
```

Comment:
At this point the tests would go like
`Translate when there is a valid response click in the field auto-translate button...`.

Adding `and the user` at the beginning would make it more readable
`Translate when there is a valid response and the user clicks in the field auto-translate button...`.

Or even better

`Translate given a valid response when the user clicks in...`